From b4b625f6dd70e43d976bcd202767e1ce5cc68e04 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:08 -0400
Subject: [ethernet] i40e: Mass conversion of smp_mb__*()

Message-id: <1409658042-22689-68-git-send-email-sassmann@redhat.com>
Patchwork-id: 90262
O-Subject: [RHEL7.1 PATCH 067/281] arch: Mass conversion of smp_mb__*()
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Peter Zijlstra <peterz@infradead.org>

Mostly scripted conversion of the smp_mb__* barriers.

Signed-off-by: Peter Zijlstra <peterz@infradead.org>
Acked-by: Paul E. McKenney <paulmck@linux.vnet.ibm.com>
Link: http://lkml.kernel.org/n/tip-55dhyhocezdw1dg7u19hmh1u@git.kernel.org
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Cc: linux-arch@vger.kernel.org
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 4e857c58efeb99393cba5a5d0d8ec7117183137c)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index efd82bf..39cef63 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -4671,7 +4671,7 @@ static void i40e_service_event_complete(struct i40e_pf *pf)
  BUG_ON(!test_bit(__I40E_SERVICE_SCHED, &pf->state));
 
  /* flush memory to make sure state is correct before next watchog */
- smp_mb__before_clear_bit();
+ smp_mb__before_atomic();
  clear_bit(__I40E_SERVICE_SCHED, &pf->state);
 }
 
-- 
1.7.1